localstack/aws/protocol/parser.py

```diff
@@ -551,28 +598,47 @@ def _get_normalized_request_uri_length(self, operation_model: OperationModel) ->
         return re.sub(r"{(.*?)}", " param", request_uri)
 
     def parse(self, request: HttpRequest) -> Tuple[OperationModel, Any]:
+        operation, path_regex = self._detect_operation(request)
+        shape: StructureShape = operation.input_shape
+        final_parsed = {}
+        if shape is not None:
+            self._parse_payload(request, shape, shape.members, path_regex, final_parsed)
+        return operation, final_parsed
+
+    def _detect_operation(self, request: HttpRequest) -> Tuple[OperationModel, Pattern[str]]:
+        """
+        Detects the operation this request is aiming for and returns the detected operation as well as the request URI
+        pattern.
+
+        :param request: to detect the operation for
+        :return: Tuple containing the detected operation the request is targeting and the URI
+                    pattern it was detected with
+        :raises: RequestParserError if the operation could not be detected
+        """
         # Use the path and the query string for the matching
         url = request.path
         if request.query_string:
             url += f"?{to_str(request.query_string)}"
         # Find the regex which matches the given path (as well as its operation)
         try:
-            path_regex, operation = next(
+            path_regex, operations = next(
                 filter(
                     lambda item: item[0].match(url),
                     self.operation_lookup[request.method].items(),
                 )
             )
+            if len(operations) > 1:
+                raise RequestParserError(
+                    f"Unable to find operation for request to service "
+                    f"{self.service.service_name}: {request.method} {request.path} "
+                    f"(ambiguous results)"
+                )
```

Comment:
I think that's too risky. This really is a case that should not happen and indicates that there is an issue in the parser or that the request really doesn't belong to the parser.
F.e. most of the skipped false-positives in the challenger branch raised this exception simply because these requests were targeting non-AWS APIs (like internal localstack endpoints).